clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -326,7 +327,7 @@ void ServiceServerLink::processNextCall()
   }
 }
 
-bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp)
+bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp, double timeout)
```

Comment:
Depending on the use case it might make sense to specify a timeout not in system time but in simtime?

Reply:
Though great part of ROS tries to keep 'kinematics' time, some other parts are strongly related to system time. We will have deadlocks when /use_sim_time=true and no /clock is provided or it is frozen.

Reply:
Would it make sense to tkae ros::Duration and WallDuration in the signature, to clarify whether the system clock or ROS time is being used?